ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java

```diff
@@ -326,82 +282,47 @@ void onAdEnd(String placement, boolean wasSuccessfulView, boolean wasCallToActio
             }
         }
 
+        @Override
+        void onAdAvailable() {
+            if (mMediationBannerListener != null) {
+                mMediationBannerListener.onAdLoaded(VungleInterstitialAdapter.this);
+            }
+        }
+
         @Override
         void onAdStart(String placement) {
             //let's load it again to mimic auto-cache, don't care about errors
-            mVungleManager.loadAd(placement, mAdConfig.getAdSize(), null);
+            if (mBannerRequest != null) {
+                mBannerRequest.preCache();
+            }
         }
 
         @Override
         void onAdFail(String placement) {
-            Log.w(TAG, "Ad playback error Placement: " + placement);
-            if (mMediationBannerListener != null) {
-                mMediationBannerListener.onAdFailedToLoad(VungleInterstitialAdapter.this,
-                        AdRequest.ERROR_CODE_INTERNAL_ERROR);
-            }
+            Log.w(TAG, "Ad playback error Placement: " + placement + ";" + mBannerRequest);
+            // might be too late to fire onAdFailedToLoad here.
+//            if (mMediationBannerListener != null) {
+//                mMediationBannerListener.onAdFailedToLoad(VungleInterstitialAdapter.this,
+//                        AdRequest.ERROR_CODE_INTERNAL_ERROR);

```

Comment:
I believe we still need to report something to admob so it can refresh asap

Reply:
ok, fixed.

Reply:
@alexander-gubov this was a request from AdMob as onAdFailedToLoad is a wrong failure reason as we have already returned success on load for this request, think theres no play/render failure event so they want us to simply log this for now

Reply:
https://github.com/googleads/googleads-mobile-android-mediation/pull/212#discussion_r385918841

Based on the review comment its asking that we could leave this in case this is ever called before onAdAvailable, I do not see how thats possible and as such think its better if we not call onAdFailedToLoad here

Reply:
Keep this code here.